### PR TITLE
cp-427: Improvements for enpoints and swagger documentation

### PR DESCRIPTION
**Pull request by OlenaYatseniuk**
- changed status code for delete chat route to 200
- changed response for delete journal route to boolean
- delete unnecessary GET journal/id route
- added pattern for createJournalEntry validation schema to allow only strings with tags to prevent UI errors when making requests from Swagger or Postman

### CODE REVIEW

**Review by v0ldemar01 — COMMENTED** [review]
@OlenaYatseniuk please check our postman config, ci postman tests fails of missing new changes there

**Review by HIMDanny — DISMISSED** [review]

File: backend/src/packages/chats/chat.controller.ts [hunk 1]
```diff
@@ -496,6 +496,13 @@ class ChatController extends BaseController {
     const { id } = options.params;
     const { id: userId } = options.user;
 
+    if (!Number(id)) {
```
Comment (v0ldemar01) [inline backend/src/packages/chats/chat.controller.ts]: Let's add it to service layer

File: backend/src/packages/journal-entries/journal-entry.controller.ts [hunk 2]
```diff
@@ -396,6 +355,13 @@ class JournalEntryController extends BaseController {
       body: { title, text },
     } = options;
 
+    if (!Number(id)) {
```
Comment (v0ldemar01) [inline backend/src/packages/journal-entries/journal-entry.controller.ts]: Same

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: shared/src/packages/journal/libs/enums/journal-entry-validation-message.enum.ts [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { JourneyEntryValidationRule } from './journal-entry-validation-rule.enum
 const JournalEntryValidationMessage = {
   TITLE_REQUIRED: 'Title is required.',
   TITLE_MUST_BE_LESS_THAN: `Title must be less than ${JourneyEntryValidationRule.MAXIMUM_TITLE_LENGTH} characters.`,
+  INVALID_TEXT_FORMAT:
+    'Invalid text format. It should start with "<p" and end with "</p>".',
```
Comment (v0ldemar01) [inline shared/src/packages/journal/libs/enums/journal-entry-validation-message.enum.ts]: Let's limit to Invalid text format, clients are not good at html tags

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: shared/src/packages/journal/libs/constants/constants.ts [hunk 4]
```diff
@@ -8,4 +8,8 @@ const NOTE_SANITIZER_OPTIONS: IOptions = {
   },
 };
 
-export { NOTE_SANITIZER_OPTIONS };
+const TEXT_PATTERN = /^<p\b[^>]*>.*<\/p>$/;
+
+const TEXT_PATTERN_NAME = 'textFormat';
```
Comment (liza-veis) [inline shared/src/packages/journal/libs/constants/constants.ts]: text pattern should be in ValidationRule, and we don't need TEXT_PATTERN_NAME

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: backend/src/packages/chats/chat.service.ts [hunk 5]
```diff
@@ -168,6 +168,13 @@ class ChatService implements Service {
     id: number;
     userId: number;
   }): Promise<boolean> {
+    if (!Number(id)) {
```
Comment (v0ldemar01) [inline backend/src/packages/chats/chat.service.ts]:
```suggestion
    if (!id)) {
```
id is already number

File: backend/src/packages/journal-entries/journal-entry.service.ts [hunk 6]
```diff
@@ -75,14 +80,23 @@ class JournalEntryService implements Service {
     title,
     text,
   }: JournalEntryUpdateRequestDto): Promise<JournalEntryGetAllItemResponseDto> {
+    if (!Number(id)) {
```
Comment (v0ldemar01) [inline backend/src/packages/journal-entries/journal-entry.service.ts]:
```suggestion
    if (!id) {
```

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: backend/src/packages/chats/chat.service.ts (hunk 5 above)
Comment (OlenaYatseniuk) [reply backend/src/packages/chats/chat.service.ts]: yes, thanks for noticing!

**Review by v0ldemar01 — APPROVED** [review]

**Pull request merged**